### PR TITLE
Feat/check arg fix

### PR DESCRIPTION
**Pull request by JoorsB**
Left in a commented function call for more debugging.

### CODE REVIEW

**Review by merlijnve — CHANGES_REQUESTED** [review]
it no worky worky

**Comment by JoorsB** [comment]
What’s the problem?

File: srcs/vm/check_args.c [hunk 1]
```diff
@@ -98,6 +99,33 @@ static void	loop_args(int argc, char **argv, t_arena *arena)
 	}
 }
 
+static void	set_champions(t_arena *arena, char **argv)
+{
+	return; // NOTE: WHAT DOES THIS DO AND WHY DOES IT WORK WITHOUT?
```
Comment (ItsWendell) [inline srcs/vm/check_args.c:104]: @JoorsB I commented this out, still works perfectly fine? Why? What should this do?

File: srcs/vm/main.c [hunk 2]
```diff
@@ -23,8 +23,9 @@ int			main(int argc, char **argv)
 	t_arena		arena;
 
 	ft_bzero(&arena, sizeof(arena));
+	arena.dump_flag = -1;
```
Comment (ItsWendell) [inline srcs/vm/main.c:26]: Waarom staat dit er in, arena is al 0, waarom -1?

File: srcs/vm/utils_2.c [hunk 3]
```diff
@@ -177,3 +177,24 @@ int		is_registry(int arg)
 		return (true);
 	return (false);
 }
+
+void		set_champ_zero(t_arena *arena, int i)
+{
+	arena->champions[i].id = 0;
```
Comment (ItsWendell) [inline srcs/vm/utils_2.c:183]: Sure this is needed? De hele arena struct wordt al bzero gedaan, de champions zijn static en worden dus ook op null gezet.

File: srcs/vm/vm.c [hunk 4]
```diff
@@ -106,7 +108,8 @@ void        vm_start(t_arena *arena_s)
     if (vm_init_cursors(arena_s) != kOk)
 		; // TODO: Error message?
     vm_introduce_champions(arena_s);
-    if (arena_s->visualizer.enabled)
+    printf("Visualzier enabled %d\n",arena_s->visualizer.enabled);
```
Comment (ItsWendell) [inline srcs/vm/vm.c]: verwijderen.

File: srcs/vm/check_args.c (hunk 1 above)
Comment (JoorsB) [reply srcs/vm/check_args.c:104]: This sets the champions in order from the first arena->champions[index];

Without this, there is a possibility with the use of "-n" that there is an empty arena->champions[index] in between used ones.    *** This was the bug with use of "-n". ***

File: srcs/vm/main.c (hunk 2 above)
Comment (JoorsB) [reply srcs/vm/main.c:26]: With [-dump nbr_cycles] nbr_cycles can be 0, I can be wrong?

File: srcs/vm/utils_2.c (hunk 3 above)
Comment (JoorsB) [reply srcs/vm/utils_2.c:183]: Yes after I relocated the struct, otherwise you have a duplicate.

**Comment by JoorsB** [comment]
If the code works without this it's fine, but there were some cases where empty structs where used. This should completely fix that for everything now and coming.

File: srcs/vm/main.c (hunk 2 above)
Comment (merlijnve) [reply srcs/vm/main.c:26]: nbr_cycles mag 0 zijn, als dump_flag niet op -1 wordt geset kan dit niet worden gecontroleerd

Comment (merlijnve) [reply srcs/vm/main.c:26]: Heb dit zelf al in check_args aangepast dus dubbelop

**Pull request merged**